torch/csrc/jit/export.cpp

```diff
@@ -751,12 +751,19 @@ bool ScriptModuleSerializer::moduleHasValidGetSetState(
   }
   auto set_schema = setstate->function().getSchema();
 
-  TORCH_CHECK(
-      set_schema.arguments().size() == 2,
-      "'__setstate__' must have 'self' and the state as its "
-      "only arguments, but found ",
-      set_schema.arguments().size(),
-      " arguments");
+  if (set_schema.arguments().size() == 1) {
+    TORCH_CHECK(
+        get_schema.returns().at(0).type()->isSubtypeOf(NoneType::get()),
+        "For '__setstate__' to have no state"
+        " param, '__getstate__' must return None");
+  } else {
+    TORCH_CHECK(
+        set_schema.arguments().size() == 2,
+        "'__setstate__' must have 'self' and the state as its "
```

Comment:
you don't check that `self` is the first argument here, maybe that's checked elsewhere

Reply:
It's a only ever a method so `self` is already installed as the first argument